### PR TITLE
Convert pkg_tar to using ctx.args to build the arg string.

### PR DESCRIPTION
**Pull request by aiuto**
This enables us to only spill to params files when we have to, saving an action most of the time.  It also opens up the possibility to always spill to params files on Windows, so that we will never run into issues with arg quoting issues.

### CODE REVIEW

**Review by jylinv0 — APPROVED** [review]

File: pkg/pkg.bzl [hunk 1]
```diff
@@ -187,25 +186,23 @@ def _pkg_tar_impl(ctx):
         )
 
     if ctx.attr.modes:
-        args += [
-            "--modes=%s=%s" % (_quote(key), ctx.attr.modes[key])
-            for key in ctx.attr.modes
-        ]
+        for key in ctx.attr.modes:
+            args.add("--modes", "%s=%s" % (_quote(key), ctx.attr.modes[key]))
```
Comment (nacl) [inline pkg/pkg.bzl:190]: Could this loop and the ones below be replaced with `args.add_all`?  Might be something we could nicely do with a lambda in Bazel 5, if/when we want to break compatibility.

File: pkg/pkg.bzl [hunk 2]
```diff
@@ -80,27 +80,26 @@ def _pkg_tar_impl(ctx):
...
+    args.add("--mode", ctx.attr.mode)
+    args.add("--owner", ctx.attr.owner)
+    args.add("--owner_name", ctx.attr.ownername)
+
     # Package dir can be specified by a file or inlined.
     if ctx.attr.package_dir_file:
         if ctx.attr.package_dir:
             fail("Both package_dir and package_dir_file attributes were specified")
-        package_dir_arg = "--directory=@" + ctx.file.package_dir_file.path
+        args.add("--directory", "@" + ctx.file.package_dir_file.path)
```
Comment (nacl) [inline pkg/pkg.bzl:95]: Do we really need to get the `path` here?  Would something like this work?

```starlark
args.add("--directory", ctx.file.package_dir_file, format = "@%s")
```

**Review by nacl — APPROVED** [review]
Some small nitty things, but otherwise LGTM.

File: pkg/pkg.bzl (hunk 1 above)
Comment (aiuto) [reply pkg/pkg.bzl:190]: `add_all` does not work for this case. It does something like 
`--modes a=b,c=d,....` but we need '--modes=a=b --modes=c=d ...`
The same for all the others.

File: pkg/pkg.bzl (hunk 2 above)
Comment (aiuto) [reply pkg/pkg.bzl:95]: That will give us the same thing. 
The rest of the code is explicit using `<f>.path` at the boundary between rule and the action. I think that is clearer than relying on the implicit conversion of file to file.path done by `%s`. 
Then there is the question of which is more efficient. The format solution might be slightly better, but this is only one arg for the action. Overall there is only the copy of this one path to save.

**Pull request merged**